api/internal/features/domain/controller/update_domain.go

```diff
@@ -41,6 +41,21 @@ func (c *DomainsController) UpdateDomain(f fuego.ContextWithBody[types.UpdateDom
 	updated, err := c.service.UpdateDomain(domainRequest.Name, user.ID.String(), domainRequest.ID)
 	if err != nil {
 		c.logger.Log(logger.Error, err.Error(), "")
+
+		if err == types.ErrDomainNotFound {
+			return nil, fuego.HTTPError{
+				Err:    err,
+				Status: http.StatusNotFound,
+			}
+		}
+
+		if err == types.ErrInvalidDomainID || err == types.ErrMissingDomainID || err == types.ErrDomainNameInvalid || err == types.ErrDomainNameTooLong || err == types.ErrDomainNameTooShort || err == types.ErrMissingDomainName {
```

Comment:
if there are too many error types that has to be used with or operator then can be simplified either by switch or slice

```
func isInvalidDomainError(err error) bool {
	switch err {
	case types.ErrInvalidDomainID,
		types.ErrMissingDomainID,
		types.ErrDomainNameInvalid,
		types.ErrDomainNameTooLong,
		types.ErrDomainNameTooShort,
		types.ErrMissingDomainName:
		return true
	default:
		return false
	}
}
```

Reply:
makes sense, done